File: backend/Backend/Models/Indicator.cs
namespace Backend.Models
{
    public class Indicator
    {
        public string IndicatorId { get; set; }   // Khóa chính
        public string Name { get; set; }          // Tên hiển thị
        public string Port { get; set; }          // COM port
        public int BaudRate { get; set; }         // Tốc độ truyền
    }
}

File: lib/models/indicator.dart
class Indicator {
  String indicatorId;
  String name;
  String port;
  int baudRate;

  Indicator({
    required this.indicatorId,
    required this.name,
    required this.port,
    required this.baudRate,
  });

  factory Indicator.fromJson(Map<String, dynamic> json) {
    return Indicator(
      indicatorId: json['indicatorId'] as String,
      name: json['name'] as String,
      port: json['port'] as String,
      baudRate: json['baudRate'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'indicatorId': indicatorId,
      'name': name,
      'port': port,
      'baudRate': baudRate,
    };
  }
}
